src/tox/venv.py

```diff
@@ -532,6 +542,31 @@ def tox_runtest(venv, redirect):
     return True  # Return non-None to indicate plugin has completed
 
 
+@tox.hookimpl
+def tox_runtest_pre(venv):
```

Comment:
I think these should be separate hooks for maximum compatibility with previous plugins implementing these (otherwise this is an added responsibility that every existing plugin that implements `tox_runtest_pre` now needs to also do!)

Reply:
The only added responsibility we're implying here is that the ``venv.status`` should be still zero after ``tox_runtest_pre``. I don't think anyone modified that until now :thinking: So what other responsibilities am I missing? (would prefer not to add new hooks, to pollute our hook namespace even more).

Reply:
ah ok, these aren't `firstresult=True` (they maybe should be? hard to say) nevermind this then

Reply:
only the commands are first result :+1: the setup and teardown (pre and post are not). And makes sense as while we don't want users to change the command evaluation; we do want to allow them to setup/teardown their plugin.